services/frontend-service/pkg/handler/handle_test.go

```diff
@@ -191,6 +208,196 @@ func TestServer_Handle(t *testing.T) {
+				Environment: "stg",
+				Config: &api.EnvironmentConfig{
+					Upstream: &api.EnvironmentConfig_Upstream{
+						Latest: &latestFlag,
+					},
+					Argocd: &api.EnvironmentConfig_ArgoCD{
+						SyncWindows: nil,
+						Destination: &api.EnvironmentConfig_ArgoCD_Destination{
+							Server:    "https://intwawiplatform-dev-aks-a17d3d29.privatelink.westeurope.azmk8s.io:443",
+							Namespace: &starFlag,
+						},
+						AccessList:             []*api.EnvironmentConfig_ArgoCD_AccessEntry{},
+						ApplicationAnnotations: nil,
+						IgnoreDifferences:      nil,
+						SyncOptions:            nil,
+					},
+					EnvironmentGroup: &starFlag,
+				},
+			},
+		},
+		{
+			name: "create environment but wrong method",
+			req: &http.Request{
+				Method: http.MethodGet,
+				URL: &url.URL{
+					Path: "/environments/stg/",
+				},
+			},
+			expectedResp: &http.Response{
+				StatusCode: http.StatusNotFound,
+			},
+			expectedBody: "unknown function ''\n",
+		},
+		{
+			name: "create environment but additional path params",
+			req: &http.Request{
+				Method: http.MethodPut,
+				URL: &url.URL{
+					Path: "/environments/stg/my-awesome-path",
+				},
+			},
+			expectedResp: &http.Response{
+				StatusCode: http.StatusNotFound,
+			},
+			expectedBody: "unknown function 'my-awesome-path'\n",
+		},
+		{
+			name:             "create environment - Azure enabled",
+			AzureAuthEnabled: true,
+			KeyRing:          exampleKeyRing,
+			req: &http.Request{
+				Method: http.MethodPost,
+				URL: &url.URL{
+					Path: "/environments/stg/",
+				},
+				Body: io.NopCloser(strings.NewReader("uncool")),
+				MultipartForm: &multipart.Form{
+					Value: map[string][]string{
+						"config":    []string{exampleConfig},
+						"signature": []string{exampleConfigSignature},
+					},
+				},
+			},
+			expectedResp: &http.Response{
+				StatusCode: http.StatusOK,
+			},
+			expectedBody: "",
+			expectedCreateEnvironmentRequest: &api.CreateEnvironmentRequest{
+				Environment: "stg",
+				Config: &api.EnvironmentConfig{
+					Upstream: &api.EnvironmentConfig_Upstream{
+						Environment: &exampleEnvironment,
+					},
+					Argocd:           nil,
+					EnvironmentGroup: nil,
+				},
+			},
+		},
+		{
+			name:             "create environment - Azure enabled - missing signature",
+			AzureAuthEnabled: true,
+			KeyRing:          exampleKeyRing,
+			req: &http.Request{
+				Method: http.MethodPost,
+				URL: &url.URL{
+					Path: "/environments/stg/",
+				},
+				Body: io.NopCloser(strings.NewReader("uncool")),
+				MultipartForm: &multipart.Form{
+					Value: map[string][]string{
+						"config": []string{exampleConfig},
+					},
+				},
+			},
+			expectedResp: &http.Response{
+				StatusCode: http.StatusBadRequest,
+			},
+			expectedBody: "Missing signature in request body",
+		},
+		{
+			name:             "create environment - Azure enabled - invalid signature",
+			AzureAuthEnabled: true,
+			KeyRing:          exampleKeyRing,
```

Comment:
please add one test for
AzureAuthEnabled: false
Signature: wrong
expected Result: Failure